exporter/datadogexporter/metadata/system/host.go

```diff
@@ -0,0 +1,52 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package system
+
+import (
+	"os"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/utils/cache"
+)
+
+type HostInfo struct {
+	OS   string
+	FQDN string
+}
+
+// GetHostInfo gets system information about the hostname
+func GetHostInfo(logger *zap.Logger) (hostInfo *HostInfo) {
+	if cacheVal, ok := cache.Get(cache.SystemHostInfoKey); ok {
```

Comment:
I think it makes sense to cache only the host name, thank you for the suggestion. I am going to use this function elsewhere in a future PR (to send a payload with all possible hostnames) but it can be handled as you suggest in that case too.

Reply:
Fixed!